apc_iterator.c

```diff
@@ -325,14 +332,20 @@ void apc_iterator_obj_init(apc_iterator_t *iterator, zval *search, zend_long for
     iterator->regex = NULL;
     iterator->search_hash = NULL;
     if (search && Z_TYPE_P(search) == IS_STRING && Z_STRLEN_P(search)) {
+		uint32_t capture_count;
 #ifdef ITERATOR_PCRE
         iterator->regex = zend_string_copy(Z_STR_P(search));
-        iterator->re = pcre_get_compiled_regex(iterator->regex, NULL, NULL);
+        iterator->re = pcre_get_compiled_regex(iterator->regex, &capture_count, NULL);
```

Comment:
pcre1 equivilent for old php is pcre_extra* in that position

Reply:
Ah, right. I've just switched this to directly use `pcre2_match_data_create_from_pattern` instead, as I don't think it makes sense to hog the global mdata for this use-case anyway, as the iterator is a bit longer-lived.